modules/core/src/main/java/org/apache/ignite/internal/streamer/StreamerSubscriber.java

```diff
@@ -241,18 +251,18 @@ private void requestMore() {
         pendingItemCount.addAndGet(count);
     }
 
-    private @Nullable Timer initFlushTimer() {
+    private void initFlushTimer() {
         int interval = options.autoFlushFrequency();
 
         if (interval <= 0) {
-            return null;
+            return;
         }
 
-        Timer timer = new Timer("client-data-streamer-flush-" + hashCode());
+        String threadPrefix = "client-data-streamer-flush-" + hashCode();
 
-        timer.schedule(new PeriodicFlushTask(), interval, interval);
+        flushTimer = Executors.newSingleThreadScheduledExecutor(new NamedThreadFactory(threadPrefix, log));
 
-        return timer;
+        flushTask = flushTimer.scheduleAtFixedRate(new PeriodicFlushTask(), interval, interval, TimeUnit.MILLISECONDS);
```

Comment:
1. Can you please explain this change?
2. PR title says "in client tests", but this code is not test

Reply:
> PR title says "in client tests", but this code is not test

Yes, but client tests were affected, because the timer sometimes didn't stop because of a race between `requestMore` and timer initialization. That's why this thread was sometimes showing up in my debugging sessions.

> Can you please explain this change?

This particular change is not related to the original problem, `Timer` is simply an older API and it is recommended to use `SheduledExecutorService` instead. I got lazy and didn't want to create a separate PR just for this.